engine/source_sym_engine.ml

```diff
@@ -136,6 +136,14 @@ let empty_group acs arity =
        rev_rows = ref [];
      }
 
+let width type_env = function
+  | Nil | Cons -> 0
+  | Int _ | Bool _ | String _ -> 1
+  | Tuple arity -> arity
+  | Variant variant_name ->
+     let k =  Source_env.ConstructorMap.find variant_name type_env |> snd in
+     List.length k.args
```

Comment:
I don't agree with this definition. In my mind the "width" of a type is its number of head constructors, and this function is supposed to compute the width of the type of a given constructor.

So:
- `Nil | Cons` should return `2` (lists have those two constructors),
- `Bool` should return `2`
- `Int | String` should return *a lot*; I guess that `max_int` would work even though it is not technically correct; what would be correct would be to return an `int option`, where `None` means that we cannot cover all cases in a pattern-matching (so always have a fallback node)
- `Tuple _arity` should return `1`
- the `Variant` case would need a different definition